src/runtime-rs/crates/shim-ctl/src/main.rs

```diff
@@ -0,0 +1,31 @@
+use runtimes::RuntimeHandlerManager;
+use common::{
+    message::Message,
+    types::{Request, ContainerConfig},
+};
+use tokio::{
+    sync::mpsc::{channel, Receiver},
+};
+
+const MESSAGE_BUFFER_SIZE: usize = 8;
+
```

Comment:
I think this is a possible test tool for further hypervisor testing and development, thus it needs some refactories.

But the refactories could be left to the PRs in the future, I am just mentioning it. I think this part is fine for now.

Reply:
Yeah, all of this is inherently WIP, and this will be true also for at least a bunch of upcoming PRs since I'd really like to do this work iteratively, out in the open so that people can point out mistakes and wrong directions early.  So there will be some rough edges inevitably - in code that is meant to be replaced by a subsequent PR.

Reply:
I agree.

Reply:
I think It could be raised as another pr, as a testing tool

Reply:
I think this could be raised as another pr, as a testing tool

Reply:
IMO it's ok to be in the same PR, since this would work as a tool for testing.

Reply:
@pmores if this is a test tool, it's better to not include this crate from `runtime-rs` crate: https://github.com/kata-containers/kata-containers/pull/5798/files#diff-50772485e77fcc2166bbc194075f08556b8c057fa21fc2be205a3b2f94ecba22R4

Reply:
At this point it's more like a dev tool although it doesn't seem clear to me at this point what will evolve from it, if anything at all. I'm fine even keeping it local if necessary. My reasoning putting it where I did was that, it's basically the same thing as the actual shim itself, just with a different way of invocation, so I put it right next to the shim.

If there is a consensus where else to put it of course I'm happy to move it although then I'd prefer to do so in a separate PR as it would mean a fairly significant rework of the final commit of this one.